source/core/DualTypeConversion.cpp

```diff
@@ -94,25 +94,24 @@ namespace Vireo
         return true;
     }
 
-    template<typename X, typename Y, typename Z> Y ConvertFromEnum(Z valueX, TypeRef typeRefY)
+    template<typename TSource, typename TDest> TDest ConvertFromEnum(TSource src, TypeRef destTypeRef)
     {
-        X integerValueX = static_cast<X>(valueX);
-        Y unsignedIntegerValueX = integerValueX >= 0 ? integerValueX : 0;
-        Y numElementsInY = static_cast<Y>(typeRefY->GetEnumItemCount());
-        if (unsignedIntegerValueX >= numElementsInY) {
-            unsignedIntegerValueX = numElementsInY - 1;
-        }
-        return unsignedIntegerValueX;
+        src = src >= 0 ? src : 0;
+        auto destEnumCount = static_cast<TDest>(destTypeRef->GetEnumItemCount());
+#pragma warning(push)
+#pragma warning(disable : 4018)  // Warning C4018 '>=': signed / unsigned mismatch
+        // It is safe to disable this warning because 'src' should be >= 0 here.
```

Comment:
Please don't put compiler-specific pragmas in the code like this unless absolutely necessary, and if you do, but #if guards to test the platform before using them.
In this case, I think it would be better to change the code to `static_cast<TDest>(src >= destEnumCount)` than to disable the warning.

Reply:
Since clang seems to ignore the pragmas, I left them in. I agree it is a good idea not to do compiler specific pragmas without guards. OTOH, casting either src or destEnumCount with other's type won't work as the C++ cast will do the normal roll-over and not stop at the target type's max value. OTOH, enum supports only UInt8,16,32, so it is safe to cast both to UInt32 to compare -- which would avoid the compiler warning. I updated the PR with this.